util/logger/logger.go

```diff
@@ -130,97 +123,112 @@ func keyvalsToFields(keyvals ...interface{}) logrus.Fields {
 				val = fmt.Sprintf("%v", hex.EncodeToString(v))
 			}
 		default:
-			val = fmt.Sprintf("%v", keyvals[i+1])
+			val = keyvals[i+1]
 		}
 		fields[key] = val
 	}
 
 	return fields
 }
 
-func (l *Logger) SetLevel(level logrus.Level) {
-	l.logger.SetLevel(level)
-}
-
-func (l *Logger) withDefaultFields() *logrus.Entry {
-	var fields logrus.Fields
-	if f, ok := l.obj.(fingerprintable); ok {
-		fields = keyvalsToFields(l.name, f.Fingerprint())
-	} else {
-		fields = keyvalsToFields("module", l.name)
+func (l *Logger) SetLevel(level zerolog.Level) {
+	l.logger = l.logger.Level(level)
+}
+
+func (l *Logger) Log(level zerolog.Level, msg string, keyvals ...interface{}) {
+	switch level {
+	case zerolog.DebugLevel:
+		l.logger.Debug().Msgf(msg, keyvalsToFields(keyvals...))
+	case zerolog.InfoLevel:
+		l.logger.Info().Msgf(msg, keyvalsToFields(keyvals...))
+	case zerolog.WarnLevel:
+		l.logger.Warn().Msgf(msg, keyvalsToFields(keyvals...))
+	case zerolog.ErrorLevel:
+		l.logger.Error().Msgf(msg, keyvalsToFields(keyvals...))
+	case zerolog.PanicLevel:
+		l.logger.Panic().Msgf(msg, keyvalsToFields(keyvals...))
+	case zerolog.TraceLevel:
+		l.logger.Trace().Msgf(msg, keyvalsToFields(keyvals...))
+	default:
+		l.logger.Log().Msgf(msg, keyvalsToFields(keyvals...))
 	}
-	return l.logger.WithFields(fields)
 }
 
-func (l *Logger) log(level logrus.Level, msg string, keyvals ...interface{}) {
-	if l.logger.IsLevelEnabled(level) {
-		l.withDefaultFields().WithFields(keyvalsToFields(keyvals...)).Log(level, msg)
-	}
-}
-
-func (l *Logger) With(keyvals ...interface{}) *logrus.Entry {
+func (l *Logger) With(keyvals ...interface{}) *zerolog.Event {
 	fields := keyvalsToFields(keyvals...)
-	return l.logger.WithFields(fields)
+	return l.logger.Log().Fields(fields)
 }
 
 func (l *Logger) Trace(msg string, keyvals ...interface{}) {
-	l.log(logrus.TraceLevel, msg, keyvals...)
+	l.Log(zerolog.TraceLevel, msg, keyvals...)
```

Comment:
I am wondering is it possible to have something like:
```
l.logger.Trace().Msgf(msg, eyvals...)
```

We get two points:
1- No need to convert keyvals to fields
2- Get rid of the switch-case

Reply:
what should I do for this?